extensions/amp-story/1.0/amp-story-grid-layer.js

```diff
@@ -108,6 +108,43 @@ export class AmpStoryGridLayer extends AmpStoryBaseLayer {
     return this.prerenderAllowed_;
   }
 
+  // TODO(26866): switch back to onMeasuredChange and remove the layoutBox
+  // equality checks.
+  /** @override */
+  onLayoutMeasure() {
+    super.onLayoutMeasure();
+
+    const format = this.element.getAttribute('format');
+    if (!format) {
+      return;
+    }
+
+    const formatSplits = format.split(':');
+    const horiz = parseInt(formatSplits[0], 10);
+    const vert = parseInt(formatSplits[1], 10);
+
+    return this.getVsync().runPromise(
+      {
+        measure: state => {
+          const parent = this.element.parentElement;
+          const vw = parent./*OK*/ clientWidth;
+          const vh = parent./*OK*/ clientHeight;
+          state.width = Math.min(vw, vh * horiz / vert);
+          state.height = Math.min(vh, vw * vert / horiz);
+        },
+        mutate: ({width, height}) => {
+          if (width === 0 && height === 0) {
+            return;
+          }
+          this.element.classList.add('i-amphtml-story-grid-template-aspect');
+          this.element.style.setProperty('--i-amphtml-story-layer-width', px(width));
```

Comment:
This won't handle resizes, right?  Since it outputs a single pixel width/height?

Reply:
Not declaratively via CSS. But this code is re-executed on resize, so it does so programmatically. The reason is that I judged if there's no right away to do the auto-resize for everything (e.g. width, height and font size), it's best to do the common denominator and change them in lock step. Some time in the future, this will be possible and we'll just switch the implementation then, with visual/format/api unchanged.